packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -192,6 +185,77 @@ export function getDeserializePrivateFunction(
   };
 }
 
+function reshapeResponse(operation: Operation, resultType: string) {
+  const response = operation.responses[0]!;
+
+  const responseProperties = getAllProperties(response.type);
+  const statements: string[] = [];
+  const visitedTypes = new Set<string>(); // Set to track visited types
+
+  statements.push(`let deserializedResponse: unknown = result.body;`);
+
+  deserialize(responseProperties, statements, visitedTypes);
+  statements.push(`return deserializedResponse as ${resultType};`);
```

Comment:
just a little curious, why we need as here ?

Reply:
Because the reshape function returns `Unknown` I was thinking to suggest adding some type magic to the function so that it could actually returns the type we need, however as I started trying it out it became too complex and likely not worth investing in it and maintaining for the internal added type-safety

Reply:
I'm fine changing `reshape` to just deal in `any` if that saves you effort :)